WalletWasabi/Crypto/Randomness/WasabiRandom.cs

```diff
@@ -23,7 +23,7 @@ public virtual byte[] GetBytes(int length)
 		public abstract int GetInt(int fromInclusive, int toExclusive);
 
 		public double NextDouble() =>
-			BitConverter.ToDouble(GetBytes(sizeof(double)), 0);
+			Math.Abs(BitConverter.ToDouble(GetBytes(sizeof(double)), 0) / double.MaxValue);
```

Comment:
wouldn't it be simpler to do this based on `GetInt`? after all timespan is basically an int type

Reply:
There is an issue in dotnet/runtime repo https://github.com/dotnet/runtime/issues/37905 and there are comments:

* https://github.com/dotnet/runtime/issues/37905#issuecomment-644205505
* https://github.com/dotnet/runtime/issues/37905#issuecomment-644248616

Based on the comments, I would expect  the implementation `NextDouble` we have has bias. I would use ints too if possible.

Reply:
given the precision of doubles, although some entropy is likely discarded by the division, even a significant amount, that's still likely to not be detectable, timing side channels over a network can extract even very slight timing differences but require repeated measurements. there's log2 of the number milliseconds in an hour is ~22, so we can probably get by with even fewer than that many bit of entropy remaining without it being an issue

Reply:
that said, i just realized this might produce NaNs, infinities, etc, which definitely is a problem

Reply:
oof, i completely missed your point that it's not the division but the sampling, sorry for being an idiot. yes this bias is a serious issue!